### ISSUE TITLE
Seeds/Peers dont update

### ISSUE THREAD

**Issue by AzukaChan** [KevinJDurant/OpenTorrentSite#5]
After the initial scrape of information from the torrent upload, seeds/peers are not updated.
The torrent will stay at the same amount of seeds/peers it had when the torrent was uploaded.

**Comment by caetaniho** [comment]
in my case everything is in 0 nothing appears even the torrent having Seeds / Peers stays at 0.
How do I fix this?

**Comment by KevinJDurant** [comment]
@caetaniho The reason you might not get any seeders/leechers statistics is because your hosting doesn't support/allow UDP requests, only HTTP requests. There aren't many torrents that still use HTTP Trackers. You need to talk to your hosting provider and ask if they can allow it.

Regarding @AzukaChan, you could always add a button where users can forcibly update the seeds and peers data. Much like how Extratorrents did it when they were still operating.

**Comment by AzukaChan** [comment]
@KevinJDurant Thanks for your reply. I didn't think you'd ever come back and reply to the messages that are being left. The button is a good idea, my original plan was to have torrents re-scraped once a user clicks on the torrent page (on a timer if possible?) just to avoid people forgetting to click the button. I feel like that would keep things more up-to-date, but it might cause more server usage depending on how many people are using the server, so in the end a button might be better.

Would you ever come back to work on this project?


@caetaniho 
> in my case everything is in 0 nothing appears even the torrent having Seeds / Peers stays at 0.
> How do I fix this?

Besides what @KevinJDurant has already said, I believe the default trackers left in BitTorrent/uTorrent are no longer available or would at least cause an error for me when testing on XAMPP, you might need to just change your trackers when making a torrent, I also had this issue until I realized it was the trackers themselves and not my host. Try using these as your trackers when making a torrent:

```
http://tracker3.itzmx.com:6961/announce

udp://exodus.desync.com:6969

http://open.acgnxtracker.com:80/announce
```

**Comment by KevinJDurant** [comment]
@AzukaChan 

> my original plan was to have torrents re-scraped once a user clicks on the torrent page (on a timer if possible?)

that idea would be fine if you're a low traffic torrent site. Just not when there's a very popular one. You don't want your IP to get blacklisted for abuse. It's up to you ofcourse. Timer would be possible, I would make an API endpoint and do an AJAX call when a user has been on the page for 1-2 minutes.

> Would you ever come back to work on this project?


Regarding your question if I ever would come back to work on this project... I feel like there are better solutions out there. This is a very basic piece of work that I made when I was still learning programming. My interests lies with other stuff like mainframes and Laravel.

I still do check this daily though :) I often checked your fork too.

I might fix something that bothers me though, the empty landing page when there are no uploaded torrents in the last 14 days. A PR is always welcome :)

**Comment by AzukaChan** [comment]
@KevinJDurant 
Thanks for your quick reply. Sad to hear you won't be coming back to work on the project, however I'm sure the few of us that use your template will manage to edit things as we see fit. 

Im currently in the process of making a delete button on the my-torrents.php so users are able to delete their torrents as well as fixing the seed/peer data not updating with either a timer or refresh button, but I don't want to upload them to my fork until its all working perfectly. However once its all there, feel free to copy it across to yours if you like ^^ once those 2 smalls things are done I would call this a perfectly working template. Have a nice day.

**Comment by caetaniho** [comment]
@AzukaChan @KevinJDurant I want to know how I make the button for the users to update the seed / peers, I'm new with php I'm learning a program if they can help me

**Comment by AzukaChan** [comment]
@caetaniho Everyone that uses this template wants the button however; there are very few people working on this template. As I previously mentioned in another thread I'm in the process of making the "update seeds/peers" button as well as a "delete torrent" button for users to remove torrents.

Once I've finished making them, they'll be available at [this fork](https://github.com/AzukaChan/OpenTorrentSite). Unless Kevin comes back to work on these things, I think waiting for an update is the best you can do.

**Comment by KevinJDurant** [comment]
@AzukaChan I am not sure if you fixed this in your pull request?

**Comment by AzukaChan** [comment]
@caetaniho 
The "delete torrent" feature has now been pulled from my fork to Kevins. So if you redownload the template from either one of us you will have a working delete button in your "My Torrents" section of the website.

As for the refresh button, I'm currently working on it and it should be ready in a few days.. I'm having trouble finding where to pull the trackers from for each torrent. If I manually add each tracker scrapeer, it will work fine; but that's not efficient and you'll just end up spamming every tracker in your list. Maybe it would be handy for the trackers to be stored separately in the database when a torrent is uploaded so they can be used later for scraping data?

I know trackers in the magnet link of the torrents but its mixed in with a bunch of other stuff and not sure how to just grab what I need and remove the rest of the magnet.

**TL;DR**
Update seeders/leechers button will arrive in a week or so.

**Comment by AzukaChan** [comment]
@KevinJDurant 
What would be the best way to get this "refresh seeders/leechers" button working on a timer? I notice BitTorrent refreshes trackers every 30 minutes on average, so maybe set the timer to allow the button to be pressed every hour just to be safe?

**Comment by KevinJDurant** [comment]
> @KevinJDurant
> What would be the best way to get this "refresh seeders/leechers" button working on a timer? I notice BitTorrent refreshes trackers every 30 minutes on average, so maybe set the timer to allow the button to be pressed every hour just to be safe?

I wouldn't really go to that length, especially for this template. I would just have the button to update the statistics immediately. I'm setting up my new development enviroment so I'll be able to push some changes.

In any case, there is always a JavaScript solution that can be applied. We can save timestamps in the browser their localstorage.

**Comment by AzukaChan** [comment]
Alright gimme your paypal; how much do I gotta pay you to fix this one up for me, @KevinJDurant ? <3

**Comment by KevinJDurant** [comment]
> Alright gimme your paypal; how much do I gotta pay you to fix this one up for me, @KevinJDurant ? <3

If only you could buy me time 😄; alright alright, I'll try and take a look at it this evening.

**Comment by KevinJDurant** [comment]
> Alright gimme your paypal; how much do I gotta pay you to fix this one up for me, @KevinJDurant ? <3

@AzukaChan there you go, it's really basic and needs more security and limitations, but this is what I could make in my little available time. Maybe you could fine tune it a bit?

**Comment by AzukaChan** [comment]
Thank you very much for helping out with the button @KevinJDurant , I had a quick look at it but sadly its not actually refreshing any data for me. I will have a look at it in the next few days to see if we can get button working once and for all :3

**Comment by AzukaChan** [comment]
This has been fixed in the latest PR.
